### PR TITLE
add change to read subjectDN from application

### PR DESCRIPTION
**Pull request by SachiniSiriwardene**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticatorTest.java [hunk 1]
```diff
@@ -57,8 +58,8 @@
 import java.util.Map;
 
 import static org.mockito.Matchers.any;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
-import static org.powermock.api.mockito.PowerMockito.when;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.*;
```
Comment (ChinthakaJ98) [inline component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticatorTest.java:62]: Shall we add the imports separately?

Comment (SachiniSiriwardene) [reply component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticatorTest.java]: fixed in https://github.com/wso2-extensions/identity-oauth-addons/pull/111/commits/c9c74e14ff756102f1db0080b516e21c2315e296

File: component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java [hunk 2]
```diff
@@ -154,6 +155,12 @@ public boolean authenticateClient(HttpServletRequest request, Map<String, List>
                 }
                 registeredCert = (X509Certificate) OAuth2Util
                         .getX509CertOfOAuthApp(oAuthClientAuthnContext.getClientId(), tenantDomain);
+                OAuthAppDO oAuthAppdo = OAuth2Util.getAppInformationByClientId(
+                        oAuthClientAuthnContext.getClientId(), tenantDomain);
+                if (StringUtils.isNotEmpty(oAuthAppdo.getTlsClientAuthSubjectDN()) &&
+                        !requestCert.getSubjectDN().toString().equals(oAuthAppdo.getTlsClientAuthSubjectDN())) {
```
Comment (janakamarasena) [inline component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java:161]: Lets invert this to make it null safe

```suggestion
                        !oAuthAppdo.getTlsClientAuthSubjectDN().equals(requestCert.getSubjectDN().toString())) {
```

Comment (SachiniSiriwardene) [reply component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java]: fixed

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**